routes/articles.js

```diff
@@ -46,11 +46,12 @@ router.post('/article/:id', requireAuth, (req, res) => {
             res.status(404)
             res.end()
         } else {
-            const { title, body, author, color } = results[0]
-            const { newTitle, newBody, newAuthor, newColor } = req.body
+            const { title, body, author, color, published } = results[0]
+            const { newTitle, newBody, newAuthor, newColor, newPublished } = req.body
+            const publishedVal = newPublished === 1 || newPublished === 0 ? newPublished : published
```

Comment:
What is this doing?

Reply:
The way we update an article is by making a POST request with the items we want to update. For example, if we want to edit the title, this is the request body:
```
{
    "newTitle": "Some New Title"
}
```

On the server (this function), we then make an UPDATE SQL command with only that field. To make it general, we update all the values, but populate the unchanged values with their old values.

This line here gets the new value for published (`publishedVal`) in this shorthand if-else statement. It is the same as:
```
let publishedVal
if (newPublished === 1 || newPublished === 0) {
    // If newPublished has a value (not null), the new value of the published column should be newPublished
    publishedVal = newPublished
} else {
    // Otherwise, the value of the published column should not be changed (same as existing)
    publishedVal = published
}
```